### ISSUE TITLE
Substitution import for testing

### ISSUE THREAD

**Issue by brianmhunt** [rollup/rollup#944]
I am having a problem with rollup in the context of including tests, reproducible in [knockout/tko.bind](knockout/tko.bind).

Specifically, `tko.bind` has tests that import modules from `devDependencies` (e.g. `tko.binding.template`), which modules in turn import `tko.bind`.  This leads to duplicate imports e.g. the function `bindingContext` (from `index.js`) also appears as `bindingContext$1` (from `node_modules/tko.bind`).

Some functions and tests fail as a result. For example, a function called `getBindingContext` has  `viewModelOrBindingContext instanceof bindingContext`, and `getBindingContext$1` has `viewModelOrBindingContext instanceof bindingContext$1`.  The `instanceof` fails for apparent reasons.

The import issue is not limited to `tko.bind`, but applies to all the `tko.*` projects in [knockout](/knockout).

I have thought of a few options.  These include:
## Option 1: Replace `tko.bind` with `../index.js`

What might make sense is to replace any imports of `tko.bind` in the `devDependencies` (e.g. `tko.binding.template`) with `../index.js` (the js:next/main export of `tko.bind`).  Unfortunately the [alias](https://github.com/frostney/rollup-plugin-alias) binding does not seem to work as expected i.e. `alias({ 'tko.bind': '../index.js' })` appears to do nothing.
## Option 2: Prioritized import from parent directory

An alternative might be to have the node-resolve plugin look in the parent directory, in addition to the `node_modules` directories.

The entire project layout looks like this:

```
/tko.utils
/tko.observable
/tko.***
/tko.bind
   /src/...
   /spec/...
   /node_modules/
      /tko.utils
      /tko.bind
      /tko.binding.template
```

This would have the additional benefit of importing from peer projects that may be in development (rather than having to include them in the project via npm).
## Option 3:  A plugin

There may be a plugin already that I haven't found yet, or worst-case that I can put together/fork from node-resolve.
## Option 4: Redesign the tests for TKO

I'm hoping we can work around the "circular"/duplicate imports with a clever Rollup config.  If we weren't working with a lot of legacy tests, this would be the best answer.
## Reproducing the issue

Here's how to see the issue at hand:
- git clone git@github.com:knockout/tko.bind.git
- cd tko.bind
- npm install
- gulp test

You can see and edit the settings passed to rollup and karma in `node_modules/tko-policy/tasks/test.js`.

If you open `localhost:7777/debug.html` in a browser you can see in the original (un-source-mapped) version of `spec/bindingAttributeBehaviors.js` the duplication of imports.

Thoughts are most welcome.

Cheers.

**Comment by brianmhunt** [comment]
Technically resolved with option #3 via a new (simple) rollup plugin https://github.com/brianmhunt/rollup-plugin-node-direct

@Rich-Harris – Please feel free to close, or comment as you may wish on any options that may be preferable.

**Comment by Rich-Harris** [comment]
I'll close this as it's fixed, but for future travellers: it's possible that the best fix for situations like these is to use the `paths` option, which allows you to do `paths: { 'tko.bind', '../index.js' }`.
